src/Kunstmaan/PagePartBundle/Controller/PagePartAdminController.php

```diff
@@ -51,17 +53,16 @@ public function newPagePartAction(Request $request)
         $pagePart      = new $pagePartClass();
 
         $formFactory = $this->container->get('form.factory');
-        $formBuilder = $formFactory->createBuilder('form');
+        $formBuilder = $formFactory->createBuilder(FormType::class);
         $pagePartAdmin->adaptForm($formBuilder);
         $id = 'newpp_' . time();
 
         $data                         = $formBuilder->getData();
         $data['pagepartadmin_' . $id] = $pagePart;
         $adminType                    = $pagePart->getDefaultAdminType();
-        if (!is_object($adminType) && is_string($adminType)) {
-            $adminType = $this->container->get($adminType);
-        }
-        $formBuilder->add('pagepartadmin_' . $id, $adminType);
+        $adminTypeFqn                 = ClassUtils::getClass($adminType);
+
+        $formBuilder->add('pagepartadmin_' . $id, $adminTypeFqn);
```

Comment:
Why? I use a form type service who needs a service:

```
    webtown_kunstmaan_chart.form.highchart_page_part.admin_type:
        class: Webtown\KunstmaanChartBundle\Form\PageParts\HighchartPagePartAdminType
        arguments:
            - '@webtown_kunstmaan_chart.configuration.parser_manager'  # <-- service!
```

I cannot set this service anywhere.

---

Otherwise I don't understand this solution and logic. There is the form type in the `$adminType` variable. Why is it wrong? Why get the FQN from the **existing** FormType then create it **again**? The `$formBuilder->add('pagepartadmin_' . $id, $adminType);` seems like a good solution for me.